react-starter-challenges-revamped/src/pages/Tic_tac_toe.tsx

```diff
@@ -1,7 +1,120 @@
 import React from "react";
+import { Alert, Card, Typography } from "@mui/material";
+import { useState } from "react";
 
-const Tic_tac_toe = () => {
-  return <div>Tic-tac-toe</div>;
+const TURNS = {
+  X: "x",
+  O: "o",
 };
+interface SquareProps {
+  children: React.ReactNode;
+  updateBoard?: (index: number) => void;
+  index?: number;
+  isSelected: boolean;
+}
 
-export default Tic_tac_toe;
+const Square = ({ children, isSelected, updateBoard, index }: SquareProps) => {
+  const className = `square ${isSelected ? "is-selected" : ""}`;
+  return (
+    <div
+      className={className}
+      onClick={() => updateBoard && index !== undefined && updateBoard(index)}
+    >
+      {children}
+    </div>
+  );
+};
+//Formas de ganar
+const WinnnerCombos = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [0, 4, 8],
+  [2, 4, 6],
+];
+
+export function Tic_tac_toe() {
+  const [board, setBoard] = useState(Array(9).fill(null));
+  const [turn, setTurn] = useState(TURNS.X);
+  const [winner, setWinner] = useState(null);
+  const [selectedIndex] = useState<number | null>(null);
+
+  //Checar si hay ganador en el tablero
+  const checkWinner = (boardToCheck: any[]) => {
+    for (const combo of WinnnerCombos) {
+      const [a, b, c] = combo;
+      if (
+        boardToCheck[a] &&
+        boardToCheck[a] === boardToCheck[b] &&
+        boardToCheck[a] === boardToCheck[c]
+      ) {
+        return boardToCheck[a];
+      }
+    }
+    return null;
+  };
+
+  const updateBoard = (index: number): void => {
+    if (board[index] || winner) return;
+    const newBoard = [...board];
+    if (!newBoard[index]) {
+      newBoard[index] = turn;
+      setBoard(newBoard);
```

Comment:
`useState` te provee con la posibilidad de pasarle un `callback` ese callback contine el valor actual del useState. Lo cual te permite manipularlo directamente y luego retornar el valor ya manipulado.

```TSX
    setBoard((prevBoard) => {
      prevBoard[index] = turn;
      return prevBoard;
    });
```

A simple vista [la documentación de react](https://react.dev/learn/updating-arrays-in-state) favorece `[...arr]`

Puede y sea preferencia personal porque para mi usando`(prevState)=>...` es mas legible 😅